src/signal-handling.c

```diff
@@ -288,6 +288,11 @@ JL_DLLEXPORT int jl_profile_is_running(void)
     return running;
 }
 
+JL_DLLEXPORT int jl_profile_is_buffer_full(void)
+{
+    return bt_size_cur + JL_BT_MAX_ENTRY_SIZE + 1 >= bt_size_max;
```

Comment:
I don't know the intent of the `+ 1` (changed in PR #34019). 
https://github.com/JuliaLang/julia/blob/852ff2bdab0b0cac987c0f1ef552ef11202cd2d5/src/stackwalk.c#L94
So based on the current implementation, I added extra `1` for the sentinel. (Note the difference between `>` and `>=`.)

Reply:
Yes, we now always add the unwind info from the stack walk (+1) to the current trace (n), then also may insert an entry with up to JL_BT_MAX_ENTRY_SIZE additional information.

This PR looks correct to me, modulo needing to be updated. Is there any particular reason you did not think so?

Reply:
I think this PR fix will work, but the intent or purpose of `+1` is not clear. I think one of the causes of the bug which this PR is trying to fix, is this unclear specification.

This is a matter of agreement between the `jl_unw_stepn` and its callers, and, of course, I am not saying #34019 is wrong.

We can reopen this PR, but I think it would be more beneficial to create a new PR than to open a stack walk lecture for me.

Reply:
BTW, I also don't understand the intent of timing control on Windows.
https://github.com/JuliaLang/julia/blob/29a6e66e4944c6d79e1f672ae4648a76599b2f21/src/signals-win.c#L340-L342

It was also weird before PR #37101, but I thought it was best effort, assuming the frequency was peaked in the first place.
PR #37101 seems to be trying to improve it by using `timeBeginPeriod`/`timeEndPeriod`. However, the unit of `nsecprof / GIGA` should be seconds. :confused:

Reply:
Ah thanks for that find! I knew the logic there had been weird, but didn't notice that the units of `timeout` were wrong also. Should be even better after https://github.com/JuliaLang/julia/pull/38208 in that case. Realistically, this just meant it always used the minimum interval (1-2ms), so hardly noticeable to most people, but worth fixing.

Reply:
Almost all looks right about that new version, if you want to make it a PR.

Note that this is using `timeout += tc.wPeriodMin` (and not `max(,1)`) since the documentation for the `Sleep` function says it returns up to that many milliseconds early. So you have to add that value to be able to set the actual desired timeout (https://docs.microsoft.com/en-us/windows/win32/sync/wait-functions#wait-functions-and-time-out-intervals)

Reply:
Oh, I'm sorry. I commented in https://github.com/JuliaLang/julia/pull/38208#issuecomment-718008334

Reply:
Now that the units issue is fixed, do you want to rebase and land that revised version of yours for this issue? Thanks!